app/src/main/java/com/nytimes/android/sample/SampleRoomStore.kt

```diff
@@ -0,0 +1,74 @@
+package com.nytimes.android.sample
+
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Database
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.PrimaryKey
+import android.arch.persistence.room.Query
+import android.arch.persistence.room.Room
+import android.arch.persistence.room.RoomDatabase
+import com.nytimes.android.external.store3.base.Fetcher
+import com.nytimes.android.external.store3.base.impl.StalePolicy
+import com.nytimes.android.external.store3.base.impl.room.RoomInternalStore
+import com.nytimes.android.external.store3.base.room.RoomPersister
+import io.reactivex.Flowable
+import io.reactivex.Observable
+import io.reactivex.Single
+
+
+// File: User.java
+@Entity
+data class User(
+        @PrimaryKey(autoGenerate = true) var uid: Int = 0,
+        val name: String,
+        val lastName: String)
+
+
+// File: UserDao.java
+@Dao
+interface UserDao {
+    @Query("SELECT name FROM user")
+    fun loadAll(): Flowable<List<String>>
+
+
+    @Insert
+    fun insertAll(vararg users: User)
+
+//    @Delete
+//    fun delete(user: User)
+}
+
+// File: AppDatabase.java
+@Database(entities = arrayOf(User::class), version = 1)
+abstract class AppDatabase : RoomDatabase() {
+    abstract fun userDao(): UserDao
+}
+
+val db = Room.databaseBuilder(SampleApp.appContext!!, AppDatabase::class.java, "db").build()
+
+val persister = object : RoomPersister<User, List<String>, String>() {
```

Comment:
A new persister needed to be created that is exposed as an observable rather than a single.  The RoomPersister takes a third param as you can have the freedom to have different queries for reading and writing. In this example writing is always done by adding a single user to the DB while read returns the list of all users. Another use case is writing a User but only reading a part of the user